doc/mgmt/gnmi/gnsi.md

```diff
@@ -0,0 +1,383 @@
+  - None
+- Scalability and performance requirements/impact
+  - There is a linear requirement for the numbers of profiles and the number of certificate revocation list rotations. The golang version of gRPC requires a complete history of CRL files since the last startup.
+- Memory requirements
+  - A map containing a few details for each profile.
+- Docker dependency
+  - None
+- Build dependency if any
+  - [gNSI](https://github.com/openconfig/gnsi/tree/main) gRPC Network Security Interface
+- Management interfaces - SNMP, CLI, RestAPI, etc.,
+  - [Configuration and management](#configuration-and-management)
+- Is this change specific to any platform? Are there dependencies for platforms to implement anything to make this feature work? If yes, explain in detail and inform community in advance.
+  - Any Platform
+- SAI API requirements, CLI requirements, ConfigDB requirements. Design is covered in following sections.
+
+#### Certz
+
+##### Certz.Rotate
+
+The Certz service definition provides the API to be used for rotating and testing PKI primitives used on network systems. The PKI primitives are grouped by SSL profiles and the service can manage multiple profiles beyond the default.
+
+The Rotate() function is a bidirectional streaming RPC which permits mutating Certificates, Root Certificate Bundles, Certificate Revocation Lists and Authentication Policies of a given profile. In a Rotate() stream it is possible to mutate one or more of the elements, and to send a Finalize message once the in-flight change has been verified to be operational. Failure to send the Finalize message will result in the candidate element being discarded and the original element being used instead.
+
+Certz.Rotate() will permit rotation, and verification of function, of any of the PKI elements. The normal use-case would be to:
+- send a PKI element to a network system as a RotateCertificateRequest.
+- verify that the services which will use the new certificate bundle continue to operate normally.
+- send a FinalizeRequest to finish the rotation process.
+
+The Certz Rotate Flow Diagram
+![Rotate Flow Chart](images/gnsi_certz_rotate.png)
+
+##### Profiles
+
+SSL profiles logically group the PKI primitives together and any gRPC Server may be configured to use a given profile. There is a default profile with the `gnxi` id that gNMI/gNOI/gNSI use. If no profile is specified, this is the profile that will be referenced.
+
+There are 3 RPC for profiles:
+- Certz.GetProfileList() - Existing profiles on a target can be discovered.
+- Certz.AddProfile() - An SSL profile can be added.
+- Certz.DeleteProfile() - When a profile is no longer needed it can be removed. Note that the gnxi SSL profile cannot be removed.
+
+##### Certificate Signing Request
+
+The server can optionally support generating a Certificate Signing Request. This allows the client to open a Certz.Rotate() stream, request a CSR and then have the Certificate signed externally.
+
+- Certz.CanGenerateCSR()
```

Comment:
Nits: maybe a description similar to all other rpcs.

Reply:
I'm not sure what more description could go here. It's just a boolean if the switch can do it or not.